server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/CassandraSessionModule.java

```diff
@@ -81,14 +83,25 @@ CassandraSessionConfiguration getCassandraSessionConfiguration(FileSystem fileSy
 
     @Provides
     @Singleton
-    Cluster provideCluster(CassandraSessionConfiguration cassandraSessionConfiguration, AsyncRetryExecutor executor) throws FileNotFoundException, ConfigurationException, ExecutionException, InterruptedException {
+    ClusterFactory.QueryLoggerConf getCassandraQueryLoggerConf() {
+        return ClusterFactory.QueryLoggerConf.builder()
+            .withConstantThreshold(500)
+            .build();
+    }
+
+    @Provides
+    @Singleton
+    Cluster provideCluster(CassandraSessionConfiguration cassandraSessionConfiguration, AsyncRetryExecutor executor, ClusterFactory.QueryLoggerConf queryLoggerConf) throws FileNotFoundException, ConfigurationException, ExecutionException, InterruptedException {
```

Comment:
Maybe I should make queryLoggerConf dependency Optional if yes how can I do that. Does "@Nullable" annotation is the proper way to do that ?